language/move-prover/bytecode/src/dataflow_analysis.rs

```diff
@@ -190,15 +190,14 @@ pub trait DataflowAnalysis: TransferFunctions {
             },
         );
         while let Some(block_id) = work_list.pop_front() {
-            let pre = state_map.remove(&block_id).expect("basic block").pre;
-            let post = self.execute_block(block_id, pre.clone(), &instrs, cfg);
+            let pre = state_map.get(&block_id).expect("basic block").pre.clone();
```

Comment:
We've been able to avoid state clone until now. Why do we need to introduce it? This can make all our analysis significantly slower.

Reply:
There was a clone even in the older code. In the  new code, the position of the clone has changed.

Reply:
Yes I understand. (See comment below.) At some point this code was designed to not have a clone (therefore the remove). So I'm wondering whether the clone is really necessary.

Reply:
Yes, it is necessary.  The cloned pre-state is passed with ownership into execute_block which returns the state after modifying it.